### PR TITLE
[23242] Avoid trying to load availableWatcher when not available

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.com/work_packages/23242/activity

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-23242-available-watcher-url.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-23242-available-watcher-url.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

File: frontend/app/components/wp-panels/watchers-panel/wp-watchers.service.js [hunk 1]
```diff
@@ -53,8 +53,15 @@ function wpWatchers($http, $q) {
   };
 
   var available = function(workPackage) {
-    var path = workPackage.links.availableWatchers.url();
-    return getWatchers(path)();
+    var link = workPackage.links.availableWatchers;
+
+    // If the user has the permission to view watchers, but not
+    // add them, this link will not be available.
+    if (link === undefined) {
```
Comment (ulferts) [inline frontend/app/components/wp-panels/watchers-panel/wp-watchers.service.js:60]: I remember @furinvader not being particularly fond of this kind of check.

You could use `angular.isUndefined` instead.

Comment (oliverguenther) [inline frontend/app/components/wp-panels/watchers-panel/wp-watchers.service.js:60]: But he was WRONG 😈  See https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/undefined for more information.

Also detailed by @romanroe in the frontend channel

**Comment by ulferts** [comment]
No error is displayed but there is still a select2 field for add watchers, which is not allowed for the user. I think that field should only be displayed if the user has that link on the resource.

**Comment by oliverguenther** [comment]
I couldn't reproduce the showing watcher select panel, however for existing watchers, the remove icon was still showing when the user didn't have the permission. That is fixed now.

**Comment by ulferts** [comment]
Sorry for the confusion about the watchers select - must have been in the wrong tab.

Works almost fine now but I would recommend to also remove the on hover border when the user lacks the remove permission. As this would entail a change in the css, I recommend to also remove/hide the button via css as you should have to pass around fewer variables.

**Comment by oliverguenther** [comment]
Okay, I've reverted the passing of the `allowedToRemove` variable and instead hide the controls on a watcher-global `-read-only` flag in css.

**Comment by ulferts** [comment]
Thx, @oliverguenther.

**Pull request merged**